umh-core/pkg/communicator/actions/set-config-file.go

```diff
@@ -0,0 +1,172 @@
+)
+
+type SetConfigFileAction struct {
+	// ─── Request metadata ────────────────────────────────────────────────────
+	userEmail    string
+	actionUUID   uuid.UUID
+	instanceUUID uuid.UUID
+
+	// ─── Plumbing ────────────────────────────────────────────────────────────
+	outboundChannel chan *models.UMHMessage
+	configManager   config.ConfigManager
+	fsService       filesystem.Service
+
+	// ─── Runtime observation ────────────────────────────────────────────────
+	systemSnapshotManager *fsm.SnapshotManager
+
+	// ─── Request payload ───────────────────────────────────────────────────
+	payload models.SetConfigFilePayload
+
+	// ─── Utilities ──────────────────────────────────────────────────────────
+	actionLogger *zap.SugaredLogger
+}
+
+// NewSetConfigFileAction creates a new SetConfigFileAction with the provided parameters.
+func NewSetConfigFileAction(userEmail string, actionUUID uuid.UUID, instanceUUID uuid.UUID, outboundChannel chan *models.UMHMessage, systemSnapshotManager *fsm.SnapshotManager, configManager config.ConfigManager) *SetConfigFileAction {
+	return &SetConfigFileAction{
+		userEmail:             userEmail,
+		actionUUID:            actionUUID,
+		instanceUUID:          instanceUUID,
+		outboundChannel:       outboundChannel,
+		systemSnapshotManager: systemSnapshotManager,
+		configManager:         configManager,
+		fsService:             configManager.GetFileSystemService(),
+		actionLogger:          logger.For(logger.ComponentCommunicator),
+	}
+}
+
+// Parse extracts the business fields from the raw JSON payload.
+func (a *SetConfigFileAction) Parse(payload interface{}) error {
+	a.actionLogger.Info("Parsing SetConfigFile payload")
+
+	// Extract SetConfigFilePayload from the interface{}
+	payloadStruct, err := ParseActionPayload[models.SetConfigFilePayload](payload)
+	if err != nil {
+		SendActionReply(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionFinishedWithFailure,
+			"Failed to parse payload as SetConfigFilePayload", a.outboundChannel, models.SetConfigFile)
+		return fmt.Errorf("failed to parse payload as SetConfigFilePayload: %w", err)
+	}
+
+	a.payload = payloadStruct
+	return nil
+}
+
+// Validate performs semantic validation of the parsed payload.
+func (a *SetConfigFileAction) Validate() error {
+	a.actionLogger.Info("Validating SetConfigFile action")
+
+	// Ensure content is not empty
+	if a.payload.Content == "" {
+		return fmt.Errorf("config file content cannot be empty")
+	}
+
+	// Ensure LastModifiedTime is not zero
+	if a.payload.LastModifiedTime == "" {
+		return fmt.Errorf("last modified time cannot be zero")
+	}
+
+	return nil
+}
+
+// Execute takes care of updating the config file content.
+func (a *SetConfigFileAction) Execute() (interface{}, map[string]interface{}, error) {
+	a.actionLogger.Info("Executing SetConfigFile action")
+
+	// Create a context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), constants.ConfigGetConfigTimeout)
```

Comment:
(major) Is it `ConfigGetConfigTimeout` or `ConfigSetConfigTimeout` ? 

If we use the same timeout constant for both `get-config` and `set-config` operation, then the constant must be renamed for clarity